src/Database/Relations/BelongsTo.php

```diff
@@ -28,25 +28,77 @@ public function __construct(Builder $query, Model $child, $foreignKey, $ownerKey
      */
     public function add(Model $model, $sessionKey = null)
```

Comment:
I think before/after Associate/Dissociate events binding would be better to place within dissociate()/associate() methods, not only to add()/remove().

Reply:
@Rike-cz fire only in associate/dissociate methods ?

Reply:
Never mind, I see the add/remove methods call the assosiate/dissociate anyway... so I'll override associate/dissociate and move the event firing there.

Reply:
There you go.